src/main/webapp/app/course/manage/overview/course-management-exercise-row.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { Course } from 'app/entities/course.model';
+import { CourseExerciseStatisticsDTO } from 'app/exercises/shared/exercise/exercise-statistics-dto.model';
+
+export enum ExerciseRowType {
+    FUTURE = 'future',
+    CURRENT = 'current',
+    PAST = 'past',
+}
+
+@Component({
+    selector: 'jhi-course-management-exercise-row',
+    templateUrl: './course-management-exercise-row.component.html',
+    styleUrls: ['course-management-exercise-row.scss'],
+})
+export class CourseManagementExerciseRowComponent implements OnInit, OnChanges {
+    @Input() course: Course;
+    @Input() exercise: Exercise;
+    @Input() statistic: CourseExerciseStatisticsDTO;
+    @Input() rowType: ExerciseRowType;
+
+    // Expose enums to the template
+    exerciseType = ExerciseType;
+    exerciseRowType = ExerciseRowType;
+
+    hasLeftoverAssessments = false;
+    isTeamExercise: boolean;
+    displayTitle: string;
+    averageScoreNumerator: number;
+
+    // TODO:
+    JSON = JSON;
+
+    getIcon(type: ExerciseType | undefined) {
```

Comment:
Why should it default to file-upload when it's undefined? Is it the most general icon? :D I guess then it makes sense

Reply:
I'm not sure which other icon we should use as default here, I don't think it matters as this won't be undefined in practice (but we want this for type checking).